src/components/InputForm.tsx

```diff
@@ -0,0 +1,160 @@
+    e.preventDefault();
+    const newEvent: Event = {
+      subject,
+      lessonDate: initialDate,
+      lessonPlan: lessonPlan || null,
+      reasonOfAbsence,
+      absentTeacherId: parseInt(absentTeacherId, 10),
+      substituteTeacherId: substituteTeacherId
+        ? parseInt(substituteTeacherId, 10)
+        : null,
+      locationId: parseInt(locationId, 10),
+    };
+
+    try {
+      const response = await fetch('/api/absence', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(newEvent),
+      });
+
+      if (response.ok) {
+        const addedEvent = await response.json();
+        onAddEvent(addedEvent);
+        setSubject('');
+        setLessonPlan('');
+        setReasonOfAbsence('');
+        setAbsentTeacherId('');
+        setSubstituteTeacherId('');
+        setLocationId('');
+        onClose();
+      } else {
+        console.error('Failed to add event:', response.statusText);
+      }
+    } catch (error) {
+      console.error('Error adding event:', error);
+    }
+  };
+
+  return (
+    <Box as="form" onSubmit={handleAddEvent} p={4}>
+      <FormControl mb={3}>
+        <FormLabel>Subject</FormLabel>
+        <Select
+          placeholder="Select subject"
+          value={subject}
+          onChange={(e) => setSubject(e.target.value)}
+          required
+          color="black"
+        >
+          <option value="M_AND_M">M_AND_M</option>
+          <option value="STRINGS">STRINGS</option>
+          <option value="CHOIR">CHOIR</option>
+          <option value="PERCUSSION">PERCUSSION</option>
+        </Select>
+      </FormControl>
+      <FormControl mb={3}>
+        <FormLabel>Lesson Plan</FormLabel>
+        <Input
+          type="text"
+          placeholder="Enter lesson plan"
+          value={lessonPlan}
+          onChange={(e) => setLessonPlan(e.target.value)}
+          color="black"
+        />
+      </FormControl>
+      <FormControl mb={3}>
+        <FormLabel>Reason of Absence</FormLabel>
+        <Input
+          type="text"
+          placeholder="Enter reason of absence"
+          value={reasonOfAbsence}
+          onChange={(e) => setReasonOfAbsence(e.target.value)}
+          required
+          color="black"
+        />
+      </FormControl>
+      <FormControl mb={3}>
+        <FormLabel>Absent Teacher ID</FormLabel>
+        <Input
+          type="number"
+          placeholder="Enter absent teacher ID"
+          value={absentTeacherId}
+          onChange={(e) => setAbsentTeacherId(e.target.value)}
+          required
+          color="black"
+        />
+      </FormControl>
+      <FormControl mb={3}>
+        <FormLabel>Substitute Teacher ID</FormLabel>
+        <Input
+          type="number"
+          placeholder="Enter substitute teacher ID"
+          value={substituteTeacherId}
+          onChange={(e) => setSubstituteTeacherId(e.target.value)}
+          color="black"
```

Comment:
nit: this might not matter because this ticket is more about frontend than styling, but on these number inputs if you scroll on your trackpad the number changes which is annoying

Reply:
doesn't seem to happen when I try to fill out the input form, do you mean like it just starts changing as soon as you scroll?